docs/en_US/NAS/retiarii/Tutorial.rst

```diff
@@ -0,0 +1,171 @@
+
+.. code-block:: python
+
+  class BlockMutator(Mutator):
+    def __init__(self, target: str):
+        super(BlockMutator, self).__init__()
+        self.target = target
+
+    def mutate(self, model):
+      nodes = model.get_nodes_by_label(self.target)
+      for node in nodes:
+        chosen_op = self.choice(candidate_op_list)
+        node.update_operation(chosen_op.type, chosen_op.params)
+
+The input of ``mutate`` is a model IR (please refer to `here <>`__ for the format and APIs of the IR), users can mutate the model with its member functions (e.g., ``get_nodes_by_label``, ``update_operation``). The mutation operations can be combined with the API ``self.choice``, in order to express a set of mutations. In the above example, the node's operation can be changed to each operation from ``candidate_op_list``.
+
+For mutator to easily target on a node (i.e., PyTorch module), we provide a placeholder module called ``nn.Placeholder``. If you want to mutate a module, you can define this module with ``nn.Placeholder``, and use mutator to mutate this placeholder to give it real computation operation.
+
+.. code-block:: python
+
+  ph = nn.Placeholder(label=f'mutable_{count}',
+    related_info={
+      'kernel_size_options': [1, 3, 5],
+      'n_layer_options': [1, 2, 3, 4],
+      'exp_ratio': exp_ratio,
+      'stride': stride
+    }
+  )
+
+``label`` is used by mutator to identify this placeholder, ``related_info`` is included in this placeholder node for mutator to get more mutation related information. A complete example code can be found in `Mnasnet base model <https://github.com/microsoft/nni/blob/master/test/retiarii_test/mnasnet/base_mnasnet.py>`__\.
+
+Explore the Defined Model Space
+-------------------------------
+
+After model space is defined, it is time to explore this model space efficiently. Users can choose proper search and training approach to explore the model space.
+
+Create a Trainer and Exploration Strategy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Classic search approach:**
+In this approach, trainer is for training each explored model, while strategy is for sampling the models. Both trainer and strategy are required to explore the model space.
+
+**Weight-sharing search approach:**
+In this approach, users only need a weight-sharing trainer, because this trainer takes charge of both search and training.
+
+In the following table, we listed the available trainers and strategies.
+
+TODO: table here.
+
+Users can write their own trainer and strategy, please refer to `here <>`__ for tutorial.
+
+Set up an Experiment
+^^^^^^^^^^^^^^^^^^^^
+
+After all the above are prepared, it is time to start an experiment to do the model search. We design unified interface for users to start their experiment. An example is shown below
+
+.. code-block:: python
+
+  exp = RetiariiExperiment(base_model, trainer, applied_mutators, simple_startegy)
+  exp_config = RetiariiExeConfig('local')
```

Comment:
import `RetiariiExperiment` and `RetiariiExeConfig`?

Reply:
no need to add, I just added the link to the complete code file.